File: Data Structures & Algorithms/duplicate-integer/submission-0.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        HashSet<Integer>set=new HashSet <>();
        for(int i=0;i<nums.length;i++){
            for(int j= i+1;j<nums.length;j++){
                if(nums[i]==nums[j])
                return true;
            }
            set.add(nums[i]);
        } 
        return false;
    }
}

File: Data Structures & Algorithms/maximum-product-subarray/submission-0.py
class Solution:
    def maxProduct(self, nums):

        res = nums[0]

        curMax = nums[0]
        curMin = nums[0]

        for i in range(1, len(nums)):

            num = nums[i]

            temp = curMax

            curMax = max(
                num,
                temp * num,
                curMin * num
            )

            curMin = min(
                num,
                temp * num,
                curMin * num
            )

            res = max(res, curMax)

        return res